### PR TITLE
outputs changes for cloud9 integration

### PR DESCRIPTION
**Pull request by snjkumar23**
## What does this PR do and why?

Changes in outputs section for cloud9 integration

## PR acceptance checklist

This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [ ] Added/Updated documentation if applicable.
- [x] Pre-commit checks passed.
- [x] Lint and Nag checks passed.
- [ ] If releasing a new version, have you bumped the version `make version part=<major|minor|patch>`?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by rezabekf — CHANGES_REQUESTED** [review]
See the inline feedback

File: content/basics/templates/outputs/index.md [hunk 1]
```diff
@@ -53,9 +52,42 @@ In this Lab, you will:
         Value: !GetAtt WebServerInstance.PublicDnsName
     ```
 
-5. Go to the AWS console and update your stack with a new template.
+1. Update the stack with new template.
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:55]:
```suggestion
1. Create the stack with a new template.
```

File: content/basics/templates/outputs/index.md [hunk 2]
```diff
@@ -53,9 +52,42 @@ In this Lab, you will:
...
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+cd cfn101-workshop/code/workspace
+:::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **MyAmiId** with  the value you have hardcoded in `resources.yaml` file earlier.
+:::code{language=shell showLineNumbers=false showCopyAction=false}
+aws cloudformation update-stack --stack-name cfn-workshop-resources --template-body file://outputs.yaml --parameters ParameterKey="AmiID",ParameterValue="MyAmiId"
+:::
+1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
+:::code{language=shell showLineNumbers=false showCopyAction=false}
+"StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-resources/739fafa0-e4d7-11ed-a000-12d9009553ff"
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:70]: Please fix to cfn-create.

Please fix the naming to `cfn-workshop-<lab-name>`

File: content/basics/templates/outputs/index.md [hunk 3]
```diff
@@ -53,9 +52,42 @@ In this Lab, you will:
...
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** link in a new tab and log in to your AWS account.
+1. Click on the stack name, for example **cfn-workshop-resources**.
+1. In the top right corner click on **Update**.
+1. In **Prepare template**, choose **Template is ready**.
+1. In **Template source**, choose **Upload a template file**.
+1. Click on **Choose file** button and navigate to your workshop directory.
+1. Select the file `outputs.yaml` and click **Next**.
+1. You can leave **Configure stack options** default, click **Next**.
+1. On the **Review <stack_name>** page, scroll down to the bottom and click on **Update stack**.
+1. You can click the **refresh** button a few times until you see in the status **UPDATE_COMPLETE**.
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:85]: Could you please update this section to create new stack. This was previously wrongly assuming to update stack where we want to create new one.

File: content/basics/templates/outputs/index.md [hunk 4]
```diff
@@ -68,34 +100,65 @@ _Outputs_ section of the template. You should continue using the `outputs.yaml`
...
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **MyAmiId** with  the value you have hardcoded in `resources.yaml` file earlier.
+  :::code{language=shell showLineNumbers=false showCopyAction=false}
+  aws cloudformation update-stack --stack-name cfn-workshop-resources --template-body file://outputs.yaml --parameters ParameterKey="AmiID",ParameterValue="MyAmiId"
+  :::
+1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
+  :::code{language=shell showLineNumbers=false showCopyAction=false}
+  "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-resources/739fafa0-e4d7-11ed-a000-12d9009553ff"
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:142]: please fix stack name `cfn-workshop-<lab-name>`

File: content/basics/templates/outputs/index.md [hunk 5]
```diff
@@ -68,34 +100,65 @@ _Outputs_ section of the template. You should continue using the `outputs.yaml`
...
+  :::
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
+1.  View the output value on the [AWS CloudFormation console](https://console.aws.amazon.com/cloudformation), in the _Outputs_ tab.
+::::
+::::tab{id="local" label="Local development"}
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** link in a new tab and log in to your AWS account.
+1. Click on the stack name, for example **cfn-workshop-resources**.
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:149]: please fix stack name `cfn-workshop-<lab-name>`

**Review by rezabekf — CHANGES_REQUESTED** [review]
Please see the comment inline. The lab should be independent from resources lab. We have previously wrongly assumed, that resources lab is completed.

File: content/basics/templates/outputs/index.md [hunk 6]
```diff
@@ -53,9 +53,44 @@ In this Lab, you will:
...
+1. Click on the stack name, for example **cfn-workshop-ec2**.
+1. In the top right corner click on **Update**.
+1. In **Prepare template**, choose **Template is ready**.
+1. In **Template source**, choose **Amazon S3 URL**.
+1. Paste the `outputs.yaml` **Object URL** you copied from the S3 bucket
+1. Click **Next**.
+1. You can leave **Configure stack options** default, click **Next**.
+1. On the **Review <stack_name>** page, scroll down to the bottom and click on **Update stack**.
+1. You can click the **refresh** button a few times until you see in the status **UPDATE_COMPLETE**.
+1. View the output value on the [AWS CloudFormation console](https://console.aws.amazon.com/cloudformation), in the _Outputs_ tab.
```
[the remark this answers is not in the document]
Comment (snjkumar23) [reply content/basics/templates/outputs/index.md]: incorporated

File: content/basics/templates/outputs/index.md (hunk 3 above)
Comment (rezabekf) [reply content/basics/templates/outputs/index.md]: I will do this.

File: content/basics/templates/outputs/index.md [hunk 7]
```diff
@@ -53,9 +52,41 @@ In this Lab, you will:
...
+
+:::::tabs{variant="container"}
+
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace`:
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+cd cfn101-workshop/code/workspace
+:::
+1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **MyAmiId** with  the value you have hardcoded in `resources.yaml` file earlier.
+:::code{language=shell showLineNumbers=false showCopyAction=false}
```
Comment (sdangol) [inline content/basics/templates/outputs/index.md:65]:
```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
```

File: content/basics/templates/outputs/index.md [hunk 8]
```diff
@@ -68,36 +99,77 @@ _Outputs_ section of the template. You should continue using the `outputs.yaml`
...
+      Value: !Ref WebServerEIP
+  ```
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace`:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you. Replace the `ParameterValue` **MyAmiId** with  the value you have hardcoded in `resources.yaml` file earlier.
+  :::code{language=shell showLineNumbers=false showCopyAction=false}
```
Comment (sdangol) [inline content/basics/templates/outputs/index.md:136]:
```suggestion
  :::code{language=shell showLineNumbers=false showCopyAction=true}
```

File: content/basics/templates/outputs/index.md [hunk 9]
```diff
@@ -68,36 +99,77 @@ _Outputs_ section of the template. You should continue using the `outputs.yaml`
...
+  :::
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
+1.  View the output value on the [AWS CloudFormation console](https://console.aws.amazon.com/cloudformation), in the _Outputs_ tab.
+::::
+::::tab{id="local" label="Local development"}
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** link in a new tab and log in to your AWS account.
+1. Click on the stack name, for example **cfn-workshop-outputs**.
+1. In the top right corner click on **Update**.
+1. In **Prepare template**, choose **Template is ready**.
+1. In **Template source**, choose **Upload a template file**.
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:151]:
```suggestion
1. In **Specify template**, choose **Upload a template file**.
```

File: content/basics/templates/outputs/index.md [hunk 10]
```diff
@@ -53,9 +52,41 @@ In this Lab, you will:
...
+:::code{language=shell showLineNumbers=false showCopyAction=false}
+"StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-outputs/739fafa0-e4d7-11ed-a000-12d9009553ff"
+:::
+ 1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
+::::
+::::tab{id="local" label="Local development"}
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** link in a new tab and log in to your AWS account.
+1. Choose **Create stack** (_With new resources (Standard)_ from the top-right side of the page.
+1. In **Prepare template**, choose **Template is ready**.
+1. In **Template source**, choose **Upload a template file**.
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:78]:
```suggestion
1. In **Specify template**, choose **Upload a template file**.
```

Comment (rezabekf) [inline content/basics/templates/outputs/index.md:78]:
```suggestion
1. In **Specify template**, choose **Upload a template file**.
```

File: content/basics/templates/outputs/index.md [hunk 11]
```diff
@@ -68,36 +99,77 @@ _Outputs_ section of the template. You should continue using the `outputs.yaml`
...
+::::tab{id="local" label="Local development"}
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** link in a new tab and log in to your AWS account.
+1. Click on the stack name, for example **cfn-workshop-outputs**.
+1. In the top right corner click on **Update**.
+1. In **Prepare template**, choose **Template is ready**.
+1. In **Template source**, choose **Upload a template file**.
+1. Click on **Choose file** button and navigate to your workshop directory.
+1. Select the file `outputs.yaml` and click **Next**.
+1. You can leave **Configure stack options** default, click **Next**.
+1. On the **Review <stack_name>** page, scroll down to the bottom and click on **Update stack**.
```
Comment (rezabekf) [inline content/basics/templates/outputs/index.md:155]:
```suggestion
1. On the **Review <stack_name>** page, scroll down to the bottom and click on **Submit**.
```

**Review by rezabekf — CHANGES_REQUESTED** [review]
Small changes required.

**Review by rezabekf — CHANGES_REQUESTED** [review]
Please resolve all the changes requested. There is quite a few required to do second review.

**Comment by snjkumar23** [comment]
> Please resolve all the changes requested. There is quite a few required to do second review.

I resolved all changes already as per your review comments, are'nt they visible in latest commit

**Comment by rezabekf** [comment]
Sorry @snjkumar23 , GitHub was down yesterday and was behaving weirdly, so I couldn't see the request has been implemented...looking now...

**Review by rezabekf — APPROVED** [review]
LGMT

**Pull request merged**